.travis.yml

```diff
@@ -1,54 +1,61 @@
-dist: xenial
-
 language: generic
-
+dist: xenial
+sudo: enabled
 os:
-  - linux
   - osx
+  - linux
 
 env:
   global:
-    - BAZEL_DOWNLOAD_PREFIX=https://github.com/bazelbuild/bazel/releases/download
-
+    - ANDROID_HOME=${HOME}/android-sdk
+    - ANDROID_SDK_TOOLS_VERSION=4333796
+    - ANDROID_BUILD_TOOLS_VERSION="28.0.3"
+    - ANDROID_PLATFORM_VERSION=28
+    - BAZELISK_VERSION="0.0.8"
+    - PATH=${PATH}:${ANDROID_HOME}/tools/bin:${HOME}/bin
   matrix:
-    - BAZEL_VERSION="0.26.1" SEGMENT="."
-    - BAZEL_VERSION="0.26.1" SEGMENT="test/test_workspace"
-    - BAZEL_VERSION="0.27.0" SEGMENT="."
-    - BAZEL_VERSION="0.27.0" SEGMENT="test/test_workspace"
-    - BAZEL_VERSION="0.27.1" SEGMENT="."
-    - BAZEL_VERSION="0.27.1" SEGMENT="test/test_workspace"
-    - BAZEL_VERSION="0.27.2" SEGMENT="."
-    - BAZEL_VERSION="0.27.2" SEGMENT="test/test_workspace"
-    - BAZEL_VERSION="0.28.0" SEGMENT="."
-    - BAZEL_VERSION="0.28.0" SEGMENT="test/test_workspace"
+    - USE_BAZEL_VERSION="0.26.1" SEGMENT="."
+    - USE_BAZEL_VERSION="0.26.1" SEGMENT="test/test_workspace"
+    - USE_BAZEL_VERSION="0.27.0" SEGMENT="."
+    - USE_BAZEL_VERSION="0.27.0" SEGMENT="test/test_workspace"
+    - USE_BAZEL_VERSION="0.27.1" SEGMENT="."
+    - USE_BAZEL_VERSION="0.27.1" SEGMENT="test/test_workspace"
+    - USE_BAZEL_VERSION="0.27.2" SEGMENT="."
+    - USE_BAZEL_VERSION="0.27.2" SEGMENT="test/test_workspace"
+    - USE_BAZEL_VERSION="0.28.0" SEGMENT="."
+    - USE_BAZEL_VERSION="0.28.0" SEGMENT="test/test_workspace"
 
 addons:
   apt:
-    sources:
-      - ubuntu-toolchain-r-test
     packages:
       - wget
-      - pkg-config
-      - zip
-      - g++
-      - zlib1g-dev
-      - unzip
-      - python3
+      - openjdk-8-jdk
   homebrew:
-    packages:
-      - wget
+    taps:
+      - adoptopenjdk/openjdk
+    casks:
+      - adoptopenjdk8
     update: true
 
 before_install:
-  - source tools/.travis.${TRAVIS_OS_NAME}
-  - wget ${BAZEL_DOWNLOAD_PREFIX}/${BAZEL_VERSION}/${BAZEL_INSTALLER}
-  - ${SHA_CMD} -c tools/${BAZEL_INSTALLER}.sha256
+  - if [ "osx" == "${TRAVIS_OS_NAME}" ]; then export PLATFORM="darwin" ; else export PLATFORM="${TRAVIS_OS_NAME}" ; fi
```

Comment:
nit: You could combine this statement with the one below, and also intend it nicely for better reading experience.

Reply:
I agree, but have had some issues with travis and spreading shell expressions across multiple lines in this yaml list, so I'm leaving it like this as a precaution.

Down the road, I will probably extract this into a sourced shell artifact.